### PR TITLE
Add params to ReadOnlySpan parameters

### PR DESCRIPTION
**Pull request by JeremyKuhne**
This adds params to ReadOnlySpan APIs primarily in System.Drawing. This enables C# to generate stack based collections in some cases.

https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-13#params-collections

This feature requires 17.10 of Visual Studio.

cc: @jaredpar 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11416)

### CODE REVIEW

**Review by jaredpar — APPROVED** [review]

**Comment by elachlan** [comment]
@JeremyKuhne The test are failing.

File: src/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs [hunk 1]
```diff
@@ -299,12 +337,29 @@ public void TransformVectors(PointF[] pts)
...
+        TransformVectors(pts.AsSpan());
+    }
 
+    /// <inheritdoc cref="TransformVectors(PointF[])"/>
+#if NET9_0_OR_GREATER
+    public
+#else
+    private
+#endif
+    void TransformVectors(ReadOnlySpan<Point> pts)
```
Comment (lonitra) [inline src/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs:361]: why don't we have `params` for this one and some of the PathGradientBrush below?

Comment (JeremyKuhne) [reply src/System.Drawing.Common/src/System/Drawing/Drawing2D/Matrix.cs]: Just missed them. Added.

File: src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs [hunk 2]
```diff
@@ -5,14 +5,35 @@
...
+#if NET9_0_OR_GREATER
+    /// <inheritdoc cref="PathGradientBrush(PointF[])"/>
+    public PathGradientBrush(params ReadOnlySpan<PointF> points) : this(WrapMode.Clamp, points) { }
+#endif
+
+    /// <summary>
+    ///  Initializes a new instance of the <see cref="PathGradientBrush"/> class with the specified points and
+    ///  <see cref="WrapMode"/>.
+    /// </summary>
+    public PathGradientBrush(PointF[] points, WrapMode wrapMode) : this(wrapMode, points.OrThrowIfNull().AsSpan()) { }
```
Comment (lonitra) [inline src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs:27]: Is params needed here for `points`?

File: src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs [hunk 3]
```diff
@@ -32,12 +53,28 @@ public PathGradientBrush(PointF[] points, WrapMode wrapMode)
...
+    /// <inheritdoc cref="PathGradientBrush(PointF[])"/>
+#if NET9_0_OR_GREATER
+    public
+#else
+    private
+#endif
+    PathGradientBrush(params ReadOnlySpan<Point> points) : this(WrapMode.Clamp, points) { }
+
+    /// <inheritdoc cref="PathGradientBrush(PointF[], WrapMode)"/>
+    public PathGradientBrush(Point[] points, WrapMode wrapMode) : this(wrapMode, points.OrThrowIfNull().AsSpan()) { }
```
Comment (lonitra) [inline src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs:68]: Ditto here

**Review by lonitra — APPROVED** [review]
🥳

**Pull request merged**